ddtrace/__init__.py

```diff
@@ -33,7 +32,6 @@
 __all__ = [
     "patch",
     "patch_all",
-    "config",
```

Comment:
Do we need to make this change to support our next gen api? 

If not, can we defer this to v5.0. This is a pretty significant change and we will need to update a lot of docs and ensure all configurations defined in `ddtrace.config` has a corresponding environment variable.